uwblueprint/rowan-house
# Sidebar progress

## Implementation Description
Brief summary of PR
Added lesson progress on the sidebar for learners, allowing them to keep track of what lesson they are and restricting them from lessons they have not yet accessed
## Screenshots
Visual depiction of PR

https://user-images.githubusercontent.com/36215359/187786725-48f11804-6b28-455f-9167-daacc8a7bb4d.mov


## What should reviewers focus on?
- When you progress through a module, lessons will be unlocked. When you go to a module, you're automatically set to the current lesson you're on, and your progress is saved.

## Testing Procedure
- Create a module with at least 2 lessons 
- Visit that module, when you finish a lesson by clicking continue the next lesson is unlocked
- Observe that locked lessons cannot be visited
- Observe that after refresh your progress is saved

## Things to Note
- Upgraded apollo from 3.3.16 to 3.6.3 to support lazy querying
- The last lesson in a module seems to not be ever in completedlessons (see the video), probably an issue in the logic elsewhere. out of scope of the ticket, will address somewhere else.

## Checklist
